src/lib/containers/MarkdownSynapse.tsx

```diff
@@ -0,0 +1,481 @@
+        let tocIdCount = 1;
+        markup = markup.replace(TOC_HEADER_REGEX, (match: string) => {
+            // replace with id so we can target them alter with click events
+            let matchWithId = `${match.substring(0, 3)} id="${tocId}${tocIdCount++}"${match.substring(3)}`;
+            return matchWithId;
+        });
+        let groups = markup.split(/(<span data-widgetparams.*?span>)/);
+        if (groups.length > 0) {
+            return this.processWidgetOrDomElement(groups, markup);
+        }
+        return
+    }
+    decodeXml(string: string) {
+        let escaped_one_to_xml_special_map = {
+            '&amp;': '&',
+            '&quot;': '"',
+            '&lt;': '<',
+            '&gt;': '>'
+        };
+        return string.replace(/(&quot;|&lt;|&gt;|&amp;)/g, function(str, item) {
+            return escaped_one_to_xml_special_map[item];
+        });
+    }
+    processWidgetMappings(rawWidgetString: string, originalMarkup: string) {
+        let widgetstringRegExp = rawWidgetString.match(/data-widgetparams=("(.*?)")/);
+        let widgetstring = this.decodeXml(widgetstringRegExp![2]);
+        let questionIndex = widgetstring.indexOf('?');
+        if (questionIndex === -1) {
+            // e.g. toc is passed, there are no params
+            return this.renderWidget(widgetstring, {}, originalMarkup);
+        }
+        let widgetType = widgetstring.substring(0, questionIndex);
+        let widgetparamsMapped = {};
+        // map out params and their values
+        widgetstring
+            .substring(questionIndex + 1)
+            .split('&')
+            .forEach(keyPair => {
+                let key, value;
+                [key, value] = keyPair.split('=');
+                value = decodeURIComponent(value);
+                widgetparamsMapped[key] = value;
+            });
+        return this.renderWidget(widgetType, widgetparamsMapped, originalMarkup);
+    }
+    processWidgetOrDomElement(widgetsToBe: string[], originalMarkup: string) {
+        let widgets = [];
+        for (let i = 0; i < widgetsToBe.length; i++) {
+            let text = widgetsToBe[i];
+            if (text.indexOf('<span data-widgetparams') !== -1) {
+                widgets.push(this.processWidgetMappings(text, originalMarkup));
+            } else {
+                widgets.push(<span key={uuidv4()} dangerouslySetInnerHTML={{ __html: text }} />);
+            }
+        }
+        return widgets;
+    }
+    renderWidget(widgetType: string, widgetparamsMapped: any, originalMarkup: string) {
+        switch (widgetType) {
+            case 'buttonlink':
+                return this.renderSynapseButton(widgetparamsMapped);
+            case 'image':
+                return this.renderSynapseImage(widgetparamsMapped);
+            case 'plot':
+                return this.renderSynapsePlot(widgetparamsMapped);
+            case 'toc':
+                return this.renderSynapseTOC(originalMarkup);
+            default:
```

Comment:
Based on wiki widget load statistics, next up should be support for:
- User/Team widget (small version of User/Team card)
- Join Team widget (complex because team may have access requirements)
- Team Member Count (easiest of them all!)